test/test_nestedtensor.py

```diff
@@ -7818,131 +7822,539 @@ def __torch_dispatch__(self, func, types, args=..., kwargs=None):
         self.assertEqual(res.shape, (4, nt.shape[1], 6))
 
 
-FORWARD_FAILURES = {
-    # === BEGIN NotImplementedError SECTION ===
-    # unary
-    "nn.functional.celu",
-    "nn.functional.elu",
-    "nn.functional.hardshrink",
-    "nn.functional.hardsigmoid",
-    "nn.functional.hardtanh",
-    "nn.functional.logsigmoid",
-    "nn.functional.mish",
-    "nn.functional.relu6",
-    "nn.functional.rrelu",
-    "nn.functional.selu",
-    "nn.functional.softplus",
-    "nn.functional.softshrink",
-    "nn.functional.threshold",
-    "rad2deg",
-    # binary
-    "__rsub__",
-    "complex",
-    "floor_divide",
-    "polar",
-    "rsub",
-    # reduction
-    "count_nonzero",
-    "linalg.vector_norm",
-    "nansum",
-    "std",
-    "std.unbiased",
-    "var",
-    "var.unbiased",
-    # === BEGIN UNSUPPORTED SECTION ===
-    # ValueError: expects strided tensor (got torch.jagged tensor)
-    "masked.amax",
-    "masked.amin",
-    "masked.argmax",
-    "masked.argmin",
-    "masked.logsumexp",
-    "masked.mean",
-    "masked.norm",
-    "masked.prod",
-    "masked.std",
-    "masked.sum",
-    "masked.var",
-    # === BEGIN BUG SECTION ===
-    # Returns a tuple of Tensors so it doesn't work with NJT's unary pointwise logic
-    "frexp",
+FORWARD_FAILURES = [
+    # not implemented
+    XFailRule(
+        error_type=NotImplementedError,
+        match_fn=lambda device, dtype, op, sample: op.full_name
+        in {
+            # unary
+            "nn.functional.celu",
+            "nn.functional.elu",
+            "nn.functional.hardshrink",
+            "nn.functional.hardsigmoid",
+            "nn.functional.hardtanh",
+            "nn.functional.logsigmoid",
+            "nn.functional.mish",
+            "nn.functional.relu6",
+            "nn.functional.rrelu",
+            "nn.functional.selu",
+            "nn.functional.softplus",
+            "nn.functional.softshrink",
+            "nn.functional.threshold",
+            "rad2deg",
+            # binary
+            "__rsub__",
+            "complex",
+            "floor_divide",
+            "polar",
+            "rsub",
+            # reduction
+            "count_nonzero",
+            "linalg.vector_norm",
+            "nansum",
+            "std",
+            "std.unbiased",
+            "var",
+            "var.unbiased",
+        },
+        name="not_implemented",
+    ),
+    # expected: masked ops don't support jagged layout
+    XFailRule(
+        error_type=ValueError,
+        error_msg="expects strided",
```

Comment:
Interesting that this doesn't give us "not_implemented". Not really a big deal, but would've been nice to have a mega xfail for the not implemented stuff lol

Reply:
yeah, super annoying :(